pkg/tests/ossm/ratelimit_test.go

```diff
@@ -33,82 +40,40 @@ var (
-		t.Fatal(err)
-	}
-
-	if _, err := util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName); err != nil {
-		t.Fatal(err)
-	}
-
-	if err := util.CheckPodRunning(meshNamespace, "app=rls"); err != nil {
-		t.Fatalf("rls deployment not ready: %v", err)
-	}
-
-	if err := util.KubeApplyContents(meshNamespace, util.RunTemplate(rateLimitFilterYaml_template, Smcp)); err != nil {
-		t.Fatalf("error applying envoy filter: %v", err)
-	}
-	util.Shell(`kubectl -n %s get envoyfilter -o yaml > rrr.yaml`, meshNamespace)
-	// util.Log.Info(msg)
-
-	// Give some time to envoy filters apply
-	time.Sleep(time.Second * 5)
-
-	host, err := util.Shell("oc -n %s get route istio-ingressgateway -o jsonpath='{.spec.host}'", meshNamespace)
-	if err != nil {
-		t.Fatalf("error getting route hostname: %v", err)
-	}
-	host = strings.Trim(host, "'")
-
-	time.Sleep(time.Duration(20) * time.Second)
-
-	// Should work first time
-	checkProductPageResponseCode(t, host, "200")
-
-	// Should fail first time
-	checkProductPageResponseCode(t, host, "429")
-
-	// Should work again after 1 minute
-	time.Sleep(time.Second * 65)
-	checkProductPageResponseCode(t, host, "200")
-
-	// Should fail
-	// time.Sleep(time.Second * 20)
-	// checkProductPageResponseCode(t, host, "429")
-}
-
-func checkProductPageResponseCode(t *testing.T, host string, expectedCode string) {
-	t.Helper()
-
-	code, err := util.Shell("curl -s -o /dev/null -w '%%{http_code}' http://%s/productpage", host)
-	if err != nil {
-		t.Fatalf("error getting productpage: %v", err)
-	}
-	code = strings.Trim(code, "'")
-	if code != expectedCode {
-		t.Fatalf("expected status code %q got %q", expectedCode, code)
-	}
+	NewTest(t).Id("T28").Groups(Full).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		namespaces := []string{"bookinfo", "redis"}
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+		smcpVersion, _ := strconv.ParseFloat(env.GetDefaultSMCPVersion(), 32)
+		if smcpVersion > 2.2 {
+			t.T().Skip("Rate limiting is not supported for SMCP versions v2.3+")
+		}
+		t.LogStep("Install Bookinfo and Redis")
+		app.InstallAndWaitReady(t, app.Bookinfo(namespaces[0]), app.Redis(namespaces[1]))
+		t.Cleanup(func() {
+			oc.DeleteNamespace(t, namespaces...)
+		})
+		t.LogStep("Patch SMCP to enable rate limiting and wait until smcp is ready")
+		oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", rateLimitSMCPPatch)
+		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+		t.LogStep("Verify rls Pod is Running")
+		rlsPod := pod.MatchingSelector("app=rls", meshNamespace)
+		oc.WaitPodRunning(t, rlsPod)
+
+		t.LogStep("Create EnvoyFilter for rate limiting")
+		oc.ApplyTemplate(t, meshNamespace, rateLimitFilterYaml_template, Smcp)
+
+		productPageURL := app.BookinfoProductPageURL(t, meshNamespace)
+		t.LogStep("Make 3 request to validate rate limit: first should work, second should fail with 429, third should work again after wait more than 10 seconds")
```

Comment:
This says 10 seconds, but the sleep is only 5 seconds.

Reply:
Done